.github/workflows/test.yaml

```diff
@@ -19,6 +19,8 @@ jobs:
           toolchain: ${{ matrix.rust }}
           override: true
           components: rustfmt
+      - name: Install cargo-hack
+        run: cargo install cargo-hack --locked
```

Comment:
Instead of running cargo-hack in the ./ci.sh file(Which requires people to have installed this and it also is going to take a little while to run), can this be put into it's own job so it runs in parallel to everything else?

Reply:
Some (one?) of the tests require the `lalrpop` binary to be built.
If a user is looking to run `./ci.sh`, I find is safe to assume that they're trying to reproduce what the CI is actually doing.
In this case building various combinaison of features/no-default-features.

`cargo-hack` does not seem to allow parallel builds.